compiler/src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -1349,6 +1349,93 @@ object Parsers {
       else
         literal(inTypeOrSingleton = true)
 
+    /** Dedent a string literal by removing common leading whitespace.
```

Comment:
For new code in the compiler we use indentation syntax and new conditional `if / then / else` syntax. The old Java conditional syntax is already disabled under `-language.future`.